sycl/source/CMakeLists.txt

```diff
@@ -346,73 +340,64 @@ set(SYCL_NON_PREVIEW_SOURCES "${SYCL_COMMON_SOURCES}"
-  # To avoid potential issues in user code we also need to provide two kinds
-  # of SYCL Runtime Library for release and debug configurations.
-  foreach(flag_var
-      CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
-      CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
-    string(REGEX REPLACE "/MDd" "" ${flag_var} "${${flag_var}}")
-    string(REGEX REPLACE "/MTd" "" ${flag_var} "${${flag_var}}")
-    string(REGEX REPLACE "/MD" "" ${flag_var} "${${flag_var}}")
-    string(REGEX REPLACE "/MT" "" ${flag_var} "${${flag_var}}")
-  endforeach()
-
-  set(WIN_DUPE "1")
-  if (SYCL_ENABLE_XPTI_TRACING)
-    add_sycl_rt_library(sycl${SYCL_MAJOR_VERSION}d sycld_object XPTI_LIB xptid COMPILE_OPTIONS "/MDd" SOURCES ${SYCL_NON_PREVIEW_SOURCES} IMPLIB_NAME sycld)
-    if(SYCL_ENABLE_MAJOR_RELEASE_PREVIEW_LIB)
-      add_sycl_rt_library(sycl${SYCL_MAJOR_VERSION}-previewd sycl-previewd_object XPTI_LIB xptid COMPILE_OPTIONS "/MDd" "/D__INTEL_PREVIEW_BREAKING_CHANGES" SOURCES ${SYCL_PREVIEW_SOURCES} IMPLIB_NAME sycl-previewd)
-    endif()
-  else()
-    add_sycl_rt_library(sycl${SYCL_MAJOR_VERSION}d sycld_object COMPILE_OPTIONS "/MDd" SOURCES ${SYCL_NON_PREVIEW_SOURCES} IMPLIB_NAME sycld)
-    if(SYCL_ENABLE_MAJOR_RELEASE_PREVIEW_LIB)
-      add_sycl_rt_library(sycl${SYCL_MAJOR_VERSION}-previewd sycl-previewd_object COMPILE_OPTIONS "/MDd" "/D__INTEL_PREVIEW_BREAKING_CHANGES" SOURCES ${SYCL_PREVIEW_SOURCES} IMPLIB_NAME sycl-previewd)
-    endif()
-  endif()
-  unset(WIN_DUPE)
-  add_library(sycld ALIAS sycl${SYCL_MAJOR_VERSION}d)
-  if(SYCL_ENABLE_MAJOR_RELEASE_PREVIEW_LIB)
-    add_library(sycl-previewd ALIAS sycl${SYCL_MAJOR_VERSION}-previewd)
-  endif()
-
-  set(SYCL_EXTRA_OPTS "/MD")
-endif()
-
 check_cxx_compiler_flag(-fsemantic-interposition HAS_SEMANTIC_INTERPOSITION_FLAG)
 if (HAS_SEMANTIC_INTERPOSITION_FLAG AND NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
   # See https://github.com/llvm/llvm-project/issues/58295.
   set_source_files_properties(device_selector.cpp PROPERTIES COMPILE_FLAGS -fsemantic-interposition)
 endif()
 
 if (WIN32)
-set(LIB_NAME "sycl${SYCL_MAJOR_VERSION}")
+  set(LIB_NAME "sycl${SYCL_MAJOR_VERSION}")
+  set(LIB_OUTPUT_NAME "${LIB_NAME}")
+  set(LIB_PREVIEW_NAME ${LIB_NAME}-preview)
+  set(LIB_PREVIEW_OUTPUT_NAME ${LIB_OUTPUT_NAME}-preview)
+  if (CMAKE_BUILD_TYPE STREQUAL Debug OR CMAKE_MSVC_RUNTIME_LIBRARY STREQUAL MultiThreadedDebugDLL)
+    set(LIB_OUTPUT_NAME "${LIB_OUTPUT_NAME}d")
+    set(LIB_PREVIEW_OUTPUT_NAME ${LIB_PREVIEW_OUTPUT_NAME}d)
+  endif()
 else()
-set(LIB_NAME "sycl")
+  set(LIB_NAME "sycl")
+  set(LIB_OUTPUT_NAME "${LIB_NAME}")
+  set(LIB_PREVIEW_NAME ${LIB_NAME}-preview)
+  set(LIB_PREVIEW_OUTPUT_NAME ${LIB_OUTPUT_NAME}-preview)
 endif()
+
 # Version-agnostic name of the import library, has effect on Windows only.
 set(IMPLIB_NAME "sycl")
+set(IMPLIB_PREVIEW_NAME "${IMPLIB_NAME}-preview")
+if (CMAKE_BUILD_TYPE STREQUAL Debug OR CMAKE_MSVC_RUNTIME_LIBRARY STREQUAL MultiThreadedDebugDLL)
```

Comment:
do we not have "if (WIN32)" or similar check here because we apply this setting in add_sycl_rt_library for windows only? can we just set IMPLIB_* vars at l348 and do changes in the win path together with other name related vars?